src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Analyzer.cs

```diff
@@ -0,0 +1,238 @@
+
+            // Retrieve the StringComparer properties that need to be flagged: CurrentCultureIgnoreCase, InvariantCultureIgnoreCase
+
+            IEnumerable<IPropertySymbol> ccicPropertyGroup = stringComparerType.GetMembers(StringComparisonCurrentCultureIgnoreCaseName).OfType<IPropertySymbol>();
+            if (!ccicPropertyGroup.Any())
+            {
+                return;
+            }
+
+            IEnumerable<IPropertySymbol> icicPropertyGroup = stringComparerType.GetMembers(StringComparisonInvariantCultureIgnoreCaseName).OfType<IPropertySymbol>();
+            if (!icicPropertyGroup.Any())
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(context =>
+            {
+                IInvocationOperation caseChangingInvocation = (IInvocationOperation)context.Operation;
+                IMethodSymbol caseChangingMethod = caseChangingInvocation.TargetMethod;
+
+                if (!caseChangingMethod.Equals(toLowerParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toLowerInvariantParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toUpperParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toUpperInvariantParameterlessMethod))
+                {
+                    return;
+                }
+
+                // Ignore parenthesized operations
+                IOperation? ancestor = caseChangingInvocation.WalkUpParentheses().WalkUpConversion().Parent;
+
+                IInvocationOperation diagnosableInvocation;
+                if (ancestor is IInvocationOperation invocationAncestor)
+                {
+                    diagnosableInvocation = invocationAncestor;
+                }
+                else if (ancestor is IArgumentOperation argumentAncestor && argumentAncestor.Parent is IInvocationOperation argumentInvocationAncestor)
+                {
+                    diagnosableInvocation = argumentInvocationAncestor;
+                }
+                else
+                {
+                    return;
+                }
+
+                IMethodSymbol diagnosableMethod = diagnosableInvocation.TargetMethod;
+
+                if (diagnosableMethod.Equals(containsStringMethod) ||
+                    diagnosableMethod.Equals(startsWithStringMethod) ||
+                    diagnosableMethod.Equals(indexOfStringMethod) ||
+                    diagnosableMethod.Equals(indexOfStringInt32Method) ||
+                    diagnosableMethod.Equals(indexOfStringInt32Int32Method))
+                {
+                    context.ReportDiagnostic(diagnosableInvocation.CreateDiagnostic(RecommendCaseInsensitiveStringComparisonRule, diagnosableMethod.Name));
+                }
+                else if (diagnosableMethod.Equals(compareToStringMethod))
+                {
+                    context.ReportDiagnostic(diagnosableInvocation.CreateDiagnostic(RecommendCaseInsensitiveStringComparerRule));
+                }
```

Comment:
Not sure if this will be a false-negative in some cases. Previously, I mentioned that the codefix produces code that is semantically different. Is it a false negative in the first place?

Or maybe report CompareTo under a separate ID to give user more control to disable it alone?

Reply:
I addressed your concern about CompareTo changing the semantics. I special-cased CompareTo so that it only triggers the analyzer but not the codefix. Check out the unit tests to see how I separate them.

Reply:
Yup, this comment is about whether we should actually warn or not, and if we should warn, whether it should be a separate ID or not.
Per https://github.com/dotnet/roslyn-analyzers/pull/6662#issuecomment-1597519573, I'm personally okay with doing this in a follow-up (unless @mavasani thinks this should be in this PR)

Reply:
Ah got it. Yes, let's continue the discussion separately.